api/include/opentelemetry/logs/logger.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+#include <chrono>
+#include <map>
+#include <vector>
+
+#include "opentelemetry/common/key_value_iterable.h"  // fix after moving to "common"
+#include "opentelemetry/logs/log_record.h"
+#include "opentelemetry/nostd/shared_ptr.h"
+#include "opentelemetry/nostd/span.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace logs
+{
+/**
+ * Handles log record creation.
+ *
+//  */
+// static const struct LogRecord defaultRecord = {
+//   timestamp : std::chrono::steady_clock::now().time_since_epoch(),
+//   trace_id : 0,
+//   span_id : 0,
+//   severity_number : 0
+// };
+
+class Logger
+{
+private:
+  std::vector<LogRecord> records;
+
+public:
+  virtual ~Logger() = default;
+
+  /* Returns the name of the logger */
+  virtual nostd::string_view getName() = 0;
+
+  /**
+   * Creates a log message with the passed characteristics.
+   *
+   * @param name the name of the log event.
+   * @param sev the severity level of the log event.
+   * @param msg the string message of the log (perhaps support std::fmt or fmt-lib format).
+   * @param record the log record (object type LogRecord) that is logged.
+   * @param attributes the attributes, stored as a 2D list of key/value pairs, that are associated
+   * with this log.
+   * @param some_obj the log object of user templated type that is logged.
+   * @throws No exceptions under any circumstances.
+   */
+
+  /* A logging statement that takes in a LogRecord.
+   * A default LogRecord that will be assigned if no parameters are passed to Logger's .log() method
+   * which should at minimum contain the trace_id, span_id, and current timestamp
+   * TODO: at the minimum, correlate timestamp, traceid, and spanid to Log Record by default
+   *
+   */
+  virtual void log(const LogRecord &record) noexcept = 0;
+
+  /** Overloaded methods for an unstructured logging **/
+  void log(nostd::string_view msg) noexcept
+  {
+    log(Severity::NONE, msg);  // Set severity to NONE as default then call the log method below
+  }
+
+  void log(Severity sev, nostd::string_view msg) noexcept
+  {
+    log(sev, msg, 0);  // Set timestamp to 0 as default then call the log method below
+  }
+
+  void log(Severity sev, nostd::string_view msg, uint64_t time) noexcept
```

Comment:
For the time: need to discuss. I personally like this. But others may tell you that they prefer `std::chrono` types - as official "clean", based on C++ standard expression of time. And are gonna be upset with time expressed as `uint64_t`.. Also when we speak about time - we'd end up detailing what units we use (nanos? millis? seconds? ticks? what epoch?).

Reply:
Thats true, will take a look at the core::SystemTimestamp's `time_since_epoch()` method.

Reply:
(minor suggestion: maybe we can keep the `uint_64` but renaming the time variable `epochTimeNano`?)